test/passes/converge_O3_metrics.bin.txt

```diff
@@ -164,19 +164,17 @@ total
    (local.get $2)
   )
   (i32.store
-   (local.tee $0
-    (i32.const 32)
-   )
+   (i32.const 32)
    (i32.const 1)
   )
-  (i32.store offset=8
-   (local.get $0)
+  (i32.store
+   (i32.const 40)
    (i32.const 2)
   )
   (drop
    (call $import$0
     (i32.const 146)
-    (local.get $0)
+    (i32.const 32)
```

Comment:
Yes, looks like `precompute-propagate` did its thing here, then `coalese-locals` and `optimize-instructions` did the rest... so this proves `--optimize-level=3` was noticed in more places, as it runs `precompute-propagate`.

In general this seems good, but we should probably investigate more here, since `-O3` is much slower than even `-O2`. And in theory it can have downsides.

I would guess the place the change happens is `optimizeAfterInlining` if it now gets the options from its parent but it didn't before. In that case, it might be better to keep it at `--optimize-level=0`, which would make this NFC. Changing that to 3 may well make sense but we should benchmark both compile time and throughput, and that seems like it could be done separately later.